sema/shacl/shaclery.py

```
@@ -0,0 +1,196 @@
+
+
+class GraphDBValidator(Validator):
+    def _gdb_validate_url(self, url: str) -> str:
+        original = urlsplit(url)
+        # wrap the path in /rest and /validate/text
+        changed = original._replace(path=f"/rest{original.path}/validate/text")
+        return changed.geturl()
+
+    def _request_conf_rpt(self, url: str, shacl_ttl_text: str) -> Graph:
+        log.debug(f"GraphDB {url=} for SHACL:\n{shacl_ttl_text}")
+        response = requests.post(
```

Ruff (0.15.6)

[error] Probable use of `requests` call without timeout

(S113)